plugin/core/registry.py

```diff
@@ -12,12 +17,56 @@
 from functools import partial
 import operator
 import sublime
+import sublime_api  # pyright: ignore[reportMissingImports]
 import sublime_plugin
 
 
 windows = WindowRegistry()
 
 
+def new_tree_view_sheet(
+    window: sublime.Window,
+    name: str,
+    data_provider: TreeDataProvider,
+    header: str = "",
+    flags: int = 0,
+    group: int = -1
+) -> Optional[TreeViewSheet]:
+    """
+    Use this function to create a new TreeView in form of a special HtmlSheet (TreeViewSheet). Only one TreeViewSheet
+    with the given name is allowed per window. If there already exists a TreeViewSheet with the same name, its content
+    will be replaced with the new data. The header argument is allowed to contain minihtml markup.
+    """
+    wm = windows.lookup(window)
+    if not wm:
+        return None
+    if name in wm.tree_view_sheets:
+        tree_view_sheet = wm.tree_view_sheets[name]
+        sheet_id = tree_view_sheet.id()
+        if tree_view_sheet.window():
+            tree_view_sheet.set_provider(data_provider, header)
+            if flags & sublime.ADD_TO_SELECTION:
+                # add to selected sheets if not already selected
+                selected_sheets = window.selected_sheets()
+                for sheet in window.sheets():
+                    if isinstance(sheet, sublime.HtmlSheet) and sheet.id() == sheet_id:
+                        if sheet not in selected_sheets:
+                            selected_sheets.append(sheet)
+                            window.select_sheets(selected_sheets)
+                        break
+            else:
+                window.focus_sheet(tree_view_sheet)
+            return tree_view_sheet
+    tree_view_sheet = TreeViewSheet(
+        sublime_api.window_new_html_sheet(window.window_id, name, "", flags, group),
```

Comment:
Why are we using internal ST API here? Doesn't the public `window.new_html_sheet` do the same?

Reply:
It's a bit difficult to explain, but the alternative would be something like
```python
_throwaway_sheet = window.new_html_sheet(name, "", flags, group)
tree_view_sheet = TreeViewSheet(_throwaway_sheet.id(), name, data_provider, header)
return tree_view_sheet
```
and because I made `TreeViewSheet` a subclass of `HtmlSheet` (so that it can use its `set_name` and `set_contents` methods, and so that things like `isinstance(HtmlSheet, ...)` works), it will also call its `__init__` method and initialize another `sublime.HtmlSheet`. So we basically have two HtmlSheet objects with the same id. I'm not sure if that would do any harm, because the only thing its (respectively the `Sheet`'s) `__init__` method does is to assign the `sheet_id` property, but that still felt wrong to me I could imagine that in theory it might be a potential cause of bugs if ST decides at some point to do more stuff in the `Sheet`'s `__init__` method (which probaly won't happen, I know, but still....).

So I basically just wanted to avoid that two `HtmlSheet` objects are created with the same id.

Reply:
Right. There would be no question if I'd notice that `window.new_html_sheet` calls `make_sheet` before returning.